services/ec2/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java

```diff
@@ -373,4 +601,44 @@ private void updateRequestDetails(Long requestId, String approverComments, Strin
                 || gatekeeperRoleService.getUserProfile().getUserId().equalsIgnoreCase(AccessRequestWrapper.getRequestorId()))
                 .collect(Collectors.toList());
     }
+
+    private Map<String, ActiveRequestUser> initializeUserMap(List<User> users) {
+        Map<String, ActiveRequestUser> userMap = new HashMap<>();
+        for (User user : users) {
+
+            ActiveRequestUser activeRequestUser = new ActiveRequestUser()
+                    .setUserId(user.getUserId().substring(3))
+                    .setGkUserId(user.getUserId())
+                    .setEmail(user.getEmail())
+                    .setActiveAccess(new ActiveAccessConsolidated().setLinux(new ArrayList<>()).setWindows(new ArrayList<>()))
+                    .setExpiredAccess(new ActiveAccessConsolidated().setLinux(new ArrayList<>()).setWindows(new ArrayList<>()));
+
+            userMap.put(user.getUserId(), activeRequestUser);
+        }
+
+        return userMap;
+    }
+
+    private List<UserNoId> findUserIntersection(List<User> userList1, List<User> userList2) {
+        Set<UserNoId> commonUsers = new HashSet<>();
+        List<User> commonList = new ArrayList<>(userList1);
+        commonList.addAll(userList2);
+        for(User user : commonList)
+            commonUsers.add(new UserNoId(user));
+        return new ArrayList<>(commonUsers);
```

Comment:
You shouldn't need most of this code, you should also copy the second object just to be safe.
```java 
 return new HashSet<>(commonList.addAll(userList2))
```

Reply:
after further discussion I am good with this change